backend/src/routes/auth.ts

```diff
@@ -160,4 +165,121 @@ export const authRoute = new Elysia({ prefix: '/auth' })
+        const formData = await request.formData();
+        const validEmaillPass = signInFormSchema.safeParse({
+            email: formData.get('email'),
+            password: formData.get('password'),
+        });
+
+        if (!validEmaillPass.success) {
+            return {
+                status: 'error',
+                errors: validEmaillPass.error.flatten().fieldErrors,
+            };
+        }
+
+        const { email, password } = validEmaillPass.data;
+
+        let user_id = '';
+
+        try {
+            const queryAuthUserData = await sql`
+            SELECT id, hashed_password
+            FROM auth_user
+            WHERE email = ${email}
+            `;
+
+            const queriedHashedPassword = queryAuthUserData[0].hashed_password;
+            user_id = queryAuthUserData[0].id;
+
+            const isPasswordMatch = await new Argon2id().verify(
+                queriedHashedPassword,
+                password,
+            );
+
+            if (!isPasswordMatch) {
+                return {
+                    status: 'error',
+                    response: {
+                        message: 'email or password is incorrect',
+                    },
+                };
+            }
+        } catch (error: any) {
+            return {
+                status: 'error',
+                response: {
+                    message: error,
+                },
+            };
+        }
+
+        const session = await lucia.createSession(user_id, {});
+        const sessionCookie = lucia.createSessionCookie(session.id);
+
+        cookie[sessionCookie.name].set({
+            value: sessionCookie.value,
+            ...sessionCookie.attributes,
+        });
+
+        return {
+            status: 'success',
+            response: {
+                message: 'login success',
+            },
+        };
+    })
+    .post('/session-check', async ({ cookie }) => {
+        const sessionID = cookie.auth_session.value;
+
+        try {
+            const sessionRecord = await sql`
+                
+                SELECT expires_at
+                FROM user_session 
+                WHERE id = ${sessionID!}
+                `;
+
+            const isSession = sessionRecord[0].expires_at;
+
+            if ( !isSession ) {
+                return {
+                    status: 'success',
+                    response: {
+                        return: true,
+                        message : "there is no session."
+                    },
+                };
+            }
+
+            const expires_at = Date.parse(sessionRecord[0].expires_at);
+            const currentTime = new Date().getTime();
+
+            if (expires_at > currentTime) {
+                return {
+                    status: 'success',
+                    response: {
+                        return: false,
+                        message : "session have not expired yet."
+                    },
+                };
+            } else {
```

Comment:
This already a guard cause why bother using else?
https://en.wikipedia.org/wiki/Guard_(computer_science)